x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -170,6 +183,76 @@ private <T> void preAnalyzeIndices(LogicalPlan parsed, ActionListener<IndexResol
         }
     }
 
+    static Set<String> fieldNames(LogicalPlan parsed) {
+        if (false == parsed.anyMatch(plan -> plan instanceof Aggregate || plan instanceof Project)) {
+            // no explicit columns selection, for example "from employees"
+            return IndexResolver.ALL_FIELDS;
+        }
+
+        Holder<Boolean> projectAll = new Holder<>(false);
+        parsed.forEachExpressionDown(UnresolvedStar.class, us -> {// explicit "*" fields selection
+            if (projectAll.get()) {
+                return;
+            }
+            projectAll.set(true);
+        });
+        if (projectAll.get()) {
+            return IndexResolver.ALL_FIELDS;
+        }
+
+        AttributeSet references = new AttributeSet();
+        // "keep" attributes are special whenever a wildcard is used in their name
+        // ie "from test | eval lang = languages + 1 | keep *l" should consider both "languages" and "*l" as valid fields to ask for
+        AttributeSet keepCommandReferences = new AttributeSet();
+
+        parsed.forEachDown(p -> {// go over each plan top-down
+            if (p instanceof RegexExtract re) { // for Grok and Dissect
+                AttributeSet dissectRefs = p.references();
+                // don't add to the list of fields the extracted ones (they are not real fields in mappings)
+                dissectRefs.subtract(new AttributeSet(re.extractedFields()));
```

Comment:
This line doesn't do anything - substract returns a new set and does NOT touch the original attributeSet.
I think you meant `dissectRefs.removeAll(re.extractedFields())`

Reply:
Nice catch. The extracted fields were removed as part of the loop down below, but I'll keep the `removeAll` call to make the loop matcher lighter.